onnxruntime/core/common/profiler.h

```diff
@@ -64,6 +64,11 @@ class Profiler {
   */
   std::string EndProfiling();
 
+  static Profiler& Instance() {
+    ORT_ENFORCE(instance_ != nullptr);
```

Comment:
If profiler does not get invoked on every run , we can lazily create the the profiler object here as well.

Reply:
It's more than just the object. Files need to be created for profiling.

---
In reply to: [269169518](https://github.com/Microsoft/onnxruntime/pull/705#discussion_r269169518) [](ancestors = 269169518)